src/lib/core/CHIPConfig.h

```diff
@@ -1339,3 +1358,14 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 /**
  * @}
  */
+
+/**
+ *  @def CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+ *
+ *  @brief
+ *    Enable persistence and resumption of subscriptions on servers.
```

Comment:
After looking at the implementation, it's not clear the subscriptions are *resumed* so much as the subscriptions are made to *appear like they never stopped*. The spec change (https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/5718)  indicated that one side would not have to kill the subscription on a single error. Here, the publisher, even after MaxInterval has passed without a report, can just send reports on the subscription (assuming the client still had the subscription). The prior subscriber does not *resume* the subscription (and cannot be forced to resume the subscription). Rather, after a reboot, this feature makes the publisher re-prime the whole subscription (or at least attempt to do so), and then stops. Unless the prior subscriber uses this opportunity to re-subscribe in a subsequent exchange (which is not logic specified anywhere right now), the priming data sent (which will be sent to ALL _N_ prior subscribers, irrespective of potential interest) will also require _N_ operational look-ups, followed by N CASE session re-establishment.

This feature has so many corner cases and assumptions that it would be important to write a small document about it and link it here (e.g. link a `.md` file in `docs/guides` or link to the header for the feature with sufficient explanations. Otherwise, the amount of knowledge that must be assumed to make use of the feature makes it basically unlikely to use by anyone but its authors.

Reply:
Filed issue #24533